pkg/virt-controller/services/template_test.go

```diff
@@ -190,6 +190,37 @@ var _ = Describe("Template", func() {
 					"]")
 				Expect(value).To(Equal(expectedIfaces))
 			})
+			It("should add default multus networks in the multus default-network annotation", func() {
+				vmi := v1.VirtualMachineInstance{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "testvmi",
+						Namespace: "default",
+						UID:       "1234",
+					},
+					Spec: v1.VirtualMachineInstanceSpec{
+						Domain: v1.DomainSpec{},
+						Networks: []v1.Network{
+							{Name: "default",
+								NetworkSource: v1.NetworkSource{
+									Multus: &v1.MultusNetwork{NetworkName: "default", Default: true},
+								}},
+							{Name: "test1",
+								NetworkSource: v1.NetworkSource{
+									Multus: &v1.MultusNetwork{NetworkName: "test1"},
+								}},
+						},
+					},
+				}
+
+				pod, err := svc.RenderLaunchManifest(&vmi)
+				Expect(err).ToNot(HaveOccurred())
+				value, ok := pod.Annotations["v1.multus-cni.io/default-network"]
+				Expect(ok).To(Equal(true))
+				Expect(value).To(Equal("default"))
+				value, ok = pod.Annotations["k8s.v1.cni.cncf.io/networks"]
+				Expect(ok).To(Equal(true))
+				Expect(value).To(Equal("[{\"interface\":\"net1\",\"name\":\"test1\",\"namespace\":\"default\"}]"))
```

Comment:
hm, shouldn't both networks be defined here? For example, if user would like to specify a custom MAC address for the default Multus network, shouldn't they have the MAC address passed through this JSON dict?

Reply:
@booxter This is not supported yet for default networks I create a PR in multus https://github.com/intel/multus-cni/pull/268 to allow default network to have a json struct an not only a name.

@booxter , @greg-bock please review it and comment there if you find that use full to get it merge.